File: Runtime/TransitionSizePrioritization.vb
'*********************************************************************************************
' ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'*********************************************************************************************

Friend Class TransitionSizePrioritization

    Private m_TransitionSizePrioritizationId As Integer
    Private m_Iteration As Nullable(Of Integer)
    Private m_Timestep As Nullable(Of Integer)
    Private m_StratumId As Nullable(Of Integer)
    Private m_TransitionGroupId As Nullable(Of Integer)
    Private m_SizePrioritization As SizePrioritization
    Private m_MaximizeFidelityToDistribution As Boolean
    Private m_MaximizeFidelityToTotalArea As Boolean

    Public Sub New(
        ByVal transitionSizePrioritizationId As Integer,
        ByVal iteration As Nullable(Of Integer),
        ByVal timestep As Nullable(Of Integer),
        ByVal stratumId As Nullable(Of Integer),
        ByVal transitionGroupId As Nullable(Of Integer),
        ByVal sizePrioritization As SizePrioritization,
        ByVal maximizeFidelityToDistribution As Boolean,
        ByVal maximizeFidelityToTotalArea As Boolean)

        Me.m_TransitionSizePrioritizationId = transitionSizePrioritizationId
        Me.m_Iteration = iteration
        Me.m_Timestep = timestep
        Me.m_StratumId = stratumId
        Me.m_TransitionGroupId = transitionGroupId
        Me.m_SizePrioritization = sizePrioritization
        Me.m_MaximizeFidelityToDistribution = maximizeFidelityToDistribution
        Me.m_MaximizeFidelityToTotalArea = maximizeFidelityToTotalArea

    End Sub

    Public ReadOnly Property TransitionSizePrioritizationId As Integer
        Get
            Return Me.m_TransitionSizePrioritizationId
        End Get
    End Property

    Public ReadOnly Property Iteration As Nullable(Of Integer)
        Get
            Return Me.m_Iteration
        End Get
    End Property

    Public ReadOnly Property Timestep As Nullable(Of Integer)
        Get
            Return Me.m_Timestep
        End Get
    End Property

    Public ReadOnly Property StratumId As Nullable(Of Integer)
        Get
            Return Me.m_StratumId
        End Get
    End Property

    Public ReadOnly Property TransitionGroupId As Nullable(Of Integer)
        Get
            Return Me.m_TransitionGroupId
        End Get
    End Property

    Public ReadOnly Property SizePrioritization As SizePrioritization
        Get
            Return Me.m_SizePrioritization
        End Get
    End Property

    Public ReadOnly Property MaximizeFidelityToDistribution As Boolean
        Get
            Return Me.m_MaximizeFidelityToDistribution
        End Get
    End Property

    Public ReadOnly Property MaximizeFidelityToTotalArea As Boolean
        Get
            Return Me.m_MaximizeFidelityToTotalArea
        End Get
    End Property

End Class

File: src/Runtime/TransitionSizePrioritization.cs
// A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2019 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

namespace SyncroSim.STSim
{
    internal class TransitionSizePrioritization
    {
        private int m_TransitionSizePrioritizationId;
        private int? m_Iteration;
        private int? m_Timestep;
        private int? m_StratumId;
        private int? m_TransitionGroupId;
        private SizePrioritization m_SizePrioritization;
        private bool m_MaximizeFidelityToDistribution;
        private bool m_MaximizeFidelityToTotalArea;

        public TransitionSizePrioritization(
            int transitionSizePrioritizationId, int? iteration, int? timestep, int? stratumId, int? transitionGroupId, 
            SizePrioritization sizePrioritization, bool maximizeFidelityToDistribution, bool maximizeFidelityToTotalArea)
        {
            this.m_TransitionSizePrioritizationId = transitionSizePrioritizationId;
            this.m_Iteration = iteration;
            this.m_Timestep = timestep;
            this.m_StratumId = stratumId;
            this.m_TransitionGroupId = transitionGroupId;
            this.m_SizePrioritization = sizePrioritization;
            this.m_MaximizeFidelityToDistribution = maximizeFidelityToDistribution;
            this.m_MaximizeFidelityToTotalArea = maximizeFidelityToTotalArea;
        }

        public int TransitionSizePrioritizationId
        {
            get
            {
                return this.m_TransitionSizePrioritizationId;
            }
        }

        public int? Iteration
        {
            get
            {
                return this.m_Iteration;
            }
        }

        public int? Timestep
        {
            get
            {
                return this.m_Timestep;
            }
        }

        public int? StratumId
        {
            get
            {
                return this.m_StratumId;
            }
        }

        public int? TransitionGroupId
        {
            get
            {
                return this.m_TransitionGroupId;
            }
        }

        public SizePrioritization SizePrioritization
        {
            get
            {
                return this.m_SizePrioritization;
            }
        }

        public bool MaximizeFidelityToDistribution
        {
            get
            {
                return this.m_MaximizeFidelityToDistribution;
            }
        }

        public bool MaximizeFidelityToTotalArea
        {
            get
            {
                return this.m_MaximizeFidelityToTotalArea;
            }
        }
    }
}
